diff --git a/packages/seacas/libraries/ioss/src/Ioss_Utils.h b/packages/seacas/libraries/ioss/src/Ioss_Utils.h
index 1df186ee24..09a2c6c5c9 100644
--- a/packages/seacas/libraries/ioss/src/Ioss_Utils.h
+++ b/packages/seacas/libraries/ioss/src/Ioss_Utils.h
@@ -20,7 +20,6 @@
 #include <cstddef> // for size_t
 #include <cstdint> // for int64_t
 #include <cstdlib> // for nullptrr
-#include <filesystem>
 #include <iostream>  // for ostringstream, etcstream, etc
 #include <stdexcept> // for runtime_error
 #include <string>    // for string
@@ -111,11 +110,7 @@ namespace Ioss {

     static bool is_path_absolute(const std::string &path)
     {
-      if (!path.empty()) {
-        std::filesystem::path p1 = path;
-        return p1.is_absolute();
-      }
-      return false;
+      return path[0] == '/';
     }

     /** \brief guess file type from extension */
