diff --git a/install-tpl.sh b/install-tpl.sh
index 5a23107a04..a592c3b3c5 100755
--- a/install-tpl.sh
+++ b/install-tpl.sh
@@ -215,7 +215,6 @@ fi
 # Check that cmake, git, wget exist at the beginning instead of erroring out later on...
 check_exec cmake
 check_exec git
-check_exec wget

 if [ "$NEEDS_SZIP" == "YES" ]
 then
@@ -423,7 +422,7 @@ then
         if [ "${H5VERSION}" == "develop" ]; then
             git clone https://github.com/HDFGroup/hdf5.git hdf5-develop
         else
-            wget --no-check-certificate https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-${hdf_base}/hdf5-${hdf_version}/src/hdf5-${hdf_version}.tar.bz2
+            curl --insecure -L -O https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-${hdf_base}/hdf5-${hdf_version}/src/hdf5-${hdf_version}.tar.bz2
         fi
         if [ "${H5VERSION}" != "develop" ]
         then
