diff --git modules/combined/test/tests/umat/tests modules/combined/test/tests/umat/tests
index 150c62752c..91cfabb865 100644
--- modules/combined/test/tests/umat/tests
+++ modules/combined/test/tests/umat/tests
@@ -12,6 +12,7 @@
                   'displacements, and temperature) as the built-in MOOSE capabilities for a '
                   'thermomechanical problem where temperature is a variable and affects material '
                   'stiffness significantly '
+    skip = "make install not supported"
   []
   [gap_heat_transfer_umat_reference]
     type = 'Exodiff'
