diff --git a/.gitmodules b/.gitmodules
index 802d79f33..b70a06a4a 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -1,9 +1,9 @@
 [submodule "contrib/metaphysicl"]
 	path = contrib/metaphysicl
-	url = ../../roystgnr/MetaPhysicL
+	url = https://github.com/roystgnr/MetaPhysicL
 [submodule "contrib/timpi"]
 	path = contrib/timpi
-	url = ../../libMesh/TIMPI
+	url = https://github.com/libMesh/TIMPI
 [submodule "m4/autoconf-submodule"]
 	path = m4/autoconf-submodule
-	url = ../../libMesh/autoconf-submodule
+	url = https://github.com/libMesh/autoconf-submodule
