diff --git a/src/tao/interface/taosolver_fg.c b/src/tao/interface/taosolver_fg.c
index d9bda5214e0..01af4c9c00e 100644
--- a/src/tao/interface/taosolver_fg.c
+++ b/src/tao/interface/taosolver_fg.c
@@ -42,7 +42,7 @@ PetscErrorCode TaoTestGradient(Tao tao, Vec x, Vec g1)
   PetscCall(PetscOptionsViewer("-tao_test_gradient_view", "View difference between hand-coded and finite difference Gradients element entries", "None", &mviewer, &format, &complete_print));
   PetscOptionsEnd();
   if (!test) {
-    if (complete_print) PetscCall(PetscViewerDestroy(&mviewer));
+    if (complete_print) PetscCall(PetscOptionsRestoreViewer(&mviewer));
     PetscFunctionReturn(PETSC_SUCCESS);
   }
 
@@ -94,7 +94,7 @@ PetscErrorCode TaoTestGradient(Tao tao, Vec x, Vec g1)
 
   if (complete_print) {
     PetscCall(PetscViewerPopFormat(mviewer));
-    PetscCall(PetscViewerDestroy(&mviewer));
+    PetscCall(PetscOptionsRestoreViewer(&mviewer));
   }
   PetscCall(PetscViewerASCIISetTab(viewer, tabs));
   PetscFunctionReturn(PETSC_SUCCESS);
